src/features/onboarding/accountCreatePin/AccountCreatePinScreen.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { useState, useEffect } from 'react'
+import { useTranslation } from 'react-i18next'
+import { RouteProp, useNavigation, useRoute } from '@react-navigation/native'
+import Text from '../../../components/Text'
+import EnterPin from '../../../assets/images/enter-pin.svg'
+import SafeAreaBox from '../../../components/SafeAreaBox'
+import PinDisplay from '../../../components/PinDisplay'
+import Keypad from '../../../components/Keypad'
+import {
+  OnboardingNavigationProp,
+  OnboardingStackParamList,
+} from '../onboardingTypes'
+
+type Route = RouteProp<OnboardingStackParamList, 'AccountCreatePinScreen'>
+const AccountCreatePinScreen = () => {
+  const { t } = useTranslation()
+  const route = useRoute<Route>()
+  const navigation = useNavigation<OnboardingNavigationProp>()
+
+  const [pin, setPin] = useState('')
+  const [failedConfirmation, setFailedConfirmation] = useState(false)
+
+  useEffect(() => {
+    if (pin.length === 6) {
+      if (route.params?.pinReset) {
+        // TODO: Nav to ConfirmResetPin
+      } else {
+        navigation.push('AccountConfirmPinScreen', {
```

Comment:
Is there a place to store these route keys as variables so they are easier to look up and keep track of?

Not sure if the new react navigation has any guidelines around that?

Reply:
For the onboarding flow there's a file `src/features/onboarding/onboardingTypes.ts` that has the types for this stack defined. You can type the `useNavigation` hook (eg. `const navigation = useNavigation<OnboardingNavigationProp>()`) and then you get type inference/checking around the routes and params. 

Here's a video that shows what I'm talking about...
https://share.getcloudapp.com/z8uYywl9

Reply:
Oh nice, no enum needed. Perfect